dbt/adapters/athena/config.py

```diff
@@ -1,12 +1,120 @@
+        is not defined, it falls back to the default timeout value. If the retrieved timeout value is less than or
+        equal to 0, a ValueError is raised as timeout must be a positive integer.
+
+        Returns:
+            int: The timeout value in seconds.
+
+        Raises:
+            ValueError: If the timeout value is not a positive integer.
+
+        """
+        timeout = self.config.get("timeout", DEFAULT_SPARK_SESSION_TIMEOUT)
+        if not isinstance(timeout, int):
+            raise TypeError("Timeout must be an integer")
+        if timeout <= 0:
+            raise ValueError("Timeout must be a positive integer")
+        LOGGER.debug(f"Setting timeout: {timeout}")
+        return timeout
+
+    def get_polling_interval(self) -> Any:
+        """
+        Get the polling interval for the configuration.
+
+        Returns:
+            Any: The polling interval value.
+
+        Raises:
+            KeyError: If the polling interval is not found in either `self.config`
+                or `self.session_kwargs`.
+        """
+        try:
+            return self.config["polling_interval"]
+        except KeyError:
+            try:
+                return self.session_kwargs["polling_interval"]
+            except KeyError:
+                return DEFAULT_POLLING_INTERVAL
+
+    def set_polling_interval(self) -> float:
+        """
+        Set the polling interval for the configuration.
+
+        Returns:
+            float: The polling interval value.
+
+        Raises:
+            ValueError: If the polling interval is not a positive integer.
+        """
+        polling_interval = self.get_polling_interval()
+        if not (isinstance(polling_interval, float) or isinstance(polling_interval, int)) or polling_interval <= 0:
+            raise ValueError(f"Polling_interval must be a positive number. Got: {polling_interval}")
+        LOGGER.debug(f"Setting polling_interval: {polling_interval}")
+        return float(polling_interval)
+
+    def set_engine_config(self) -> Dict[str, Any]:
+        """Set the engine configuration.
+
+        Returns:
+            Dict[str, Any]: The engine configuration.
+
+        Raises:
+            TypeError: If the engine configuration is not of type dict.
+            KeyError: If the keys of the engine configuration dictionary do not match the expected format.
+        """
+        engine_config = self.config.get(
+            "engine_config",
+            {
+                "CoordinatorDpuSize": DEFAULT_SPARK_COORDINATOR_DPU_SIZE,
+                "MaxConcurrentDpus": DEFAULT_SPARK_MAX_CONCURRENT_DPUS,
+                "DefaultExecutorDpuSize": DEFAULT_SPARK_EXECUTOR_DPU_SIZE,
+            },
+        )
+        if not isinstance(engine_config, dict):
+            raise TypeError("Engine configuration has to be of type dict")
+
+        expected_keys = {"CoordinatorDpuSize", "MaxConcurrentDpus", "DefaultExecutorDpuSize"}
```

Comment:
@Avinash-1394 fyi added a comment in the resolved thread below: 
https://github.com/dbt-athena/dbt-athena/pull/248/files#r1436610932

Reply:
Hey @sankeerthnagapuri,

Thank you so much for that! Did you run into any issues while testing the branch? I only ran it with 7 models and I didn't experiment with complex models. I can see some scenarios where long running models might cause issues. Curious if you have any suggestions for improvement :)

Reply:
Hi @Avinash-1394 

I just got to know that the engine configuration is correct for the current boto3 version 1.26.
I had to make a few changes to support iceberg table creation and I can send a PR to your branch with those changes

But yeah if we have to support non-hive tables (iceberg, hudi, delta lake) we should consider upgrading boto3 package and then allow SparkProperties for engine config.

The use case where I used this branch is to create an iceberg table from a hive table which has terabytes of data. The issues I faced with trino Athena are:
1. General capacity didnt work for the amount of data. We used reserved capacity.
2. We had 104 partitions which is above 100 limit per query for Trino/Presto. So, tried lowering the number of partitions to test it. It failed with timeout of 6 hours (soft limit by aws).
3. Finally we ran a loop command for each partition and it ran for more than 19 hours for processing all batches.
And thats a lot of time, so I tried using this branch and with few changes to support iceberg table, I was able to create iceberg table in a single query and in under 4 hours run time. Although we cannot use this on production but pyspark has huge potential to solve the limitations of Trino especially for larger datasets.

Reply:
and I didn't want to use the auto batch processing option made available  in this dbt-athena package as that would only start the batch after a failed run and if a run takes 6 hours to fail (for big data use cases), its a lot of money and time spent before the actual batch process would begin.

Reply:
@Avinash-1394 I raised a PR to your repository. 
https://github.com/Avinash-1394/dbt-athena-1/pull/1
Please let me know if you have any questions related to the changes made in that PR to be able to create iceberg tables through spark

Reply:
Thank you so much for that PR. I have raised some comments. Will try to get that merged in this week!

Reply:
thank you for the review and merging!